### DISCUSSION TITLE
(DEV) Export & (PROD) Import

### DISCUSSION THREAD

**Discussion by AkosBakos** [Micke-K/IntuneManagement#81] [Q&A] [answered]
What could be the best way to import all of existing profiles, settings catalog, like we did during the pilot phase. And import these for the production purpose.

The question is either, how could be easily rename the exported DEV items during the import phase?
Plane B, we are going throw all of the imported items and rename it manually, which is a bigger environment takes a bit more time.

**Comment by Micke-K** [comment, marked as answer]
So these are different tenants? How is the naming standard eg what are you renaming?

You could create a script to rename by changing the name value in the json files. This _MIGHT_ break a few things though depending on the policy type you use. Some policies uses a reference value that is not an actual property. The script will create a an additional property for this that is uses during import but that will be name based. So renaming policies can break a reference.

In these scenarios is also good practice to have no environment reference in the description since that would be one additional thing you have to change.

Another thing you could do is to use Bulk Copy and rename them. If prod is in a different tenant, you export the prod policies with a name filter and then import them in the prod tenant. Then use Bulk Delete to clean up these "prod" polices  from the Dev tenant. 

Another thing to consider is ScopeTags, if you use for environment tagging. This could be cause an issue if they are renamed.

No matter how you do it, I'd recommend verifying all values after import, especially policies with a certificate reference.

And one last last thing. Importing without assignment is a safe option but might give you some manual work. If you import with assignments, I'd be careful if you have any devices in prod. And remember that this will create Azure AD groups if they don't exist.

Cheers!

**Reply by AkosBakos** [reply]
G'day, mate! :-)

Thank you so much for writing such a nice summary, tips, tricks etc. from your own perspective and about the possibilities

Cheers

**Reply by Micke-K** [reply]
No worries mate :)

Let me know how it goes or if you have any other questions. 

Cheers and good luck!
